pallets/lbp/src/lib.rs

```diff
@@ -658,7 +663,13 @@ pub mod pallet {
 
 			Self::deposit_event(Event::LiquidityRemoved(pool_id, asset_a, asset_b, amount_a, amount_b));
 
-			Ok(())
+			 let amount_a_perbill: Balance = transfer_to_xyk * amount_a;
+			 let amount_b_perbill: Balance = transfer_to_xyk * amount_b;
```

Comment:
I don't see any weight checks here. You're assuming the pool ends at 50:50? (this doesn't need to be the case). If you do this at non-50:50 the resulting pool will have a wrong price

Reply:
updated